Dockerfile

```diff
@@ -0,0 +1,33 @@
+FROM registry.fedoraproject.org/fedora-minimal:36
+
+RUN microdnf \
+      --assumeyes \
+      --nodocs \
+      --setopt=install_weak_deps=0 \
+      install \
+        acpica-tools \
+        g++ \
+        gcc \
+        gcc-aarch64-linux-gnu \
+        gcc-arm-linux-gnu \
+        gcc-riscv64-linux-gnu \
+        git \
+        libX11-devel \
+        libXext-devel \
+        libuuid-devel \
+        make \
+        nuget \
+        nasm \
+        python \
+        python3-distutils-extra \
+        python3-pip \
+        python3-setuptools \
+        qemu-system-aarch64-core \
```

Comment:
Do we know how much size the QEMU packages add? I'm inclined to say this is the type of thing we should create a second layer container for. i.e. have fedora-36 & fedora-36-qemu. What do you think?

Reply:
Sure we can split the image. I created a monolithic one for
simplicity, to get started.

For testing, I split the Dockerfile into four, building onto of
each other. This way I get the following sizes:

| Layer     | Total [MB] | Delta [MB] |
| :---      | ---:       | ---:       |
| Base + Compilers | 975        |            |
| Python    | 1070       | 95         |
| NuGet     | 1250       | 180        |
| Qemus     | 1740       | 490        |

If we split off Qemu, then we have to do some work on
the pipeline jobs as well; split the jobs that build and
test into two, and hand over the firmware images as artifacts.


An orthogonal approach would be to split by architecture.

Reply:
Quick test of splitting by architecture with a common base image:

| Image   | Total [MB] | Delta to base [MB] |
| :---    |       ---: |               ---: |
| Base    | 846        |                    |
| x86     | 953        | 107                |
| arm     | 1180       | 334                |
| aarch64 | 1260       | 414                |
| riscv64 | 976        | 130                |

Note: Docker shows image sizes as including all base images/layers (so what one would have to pull if the local registry is empty).


The base image contains python, make, git, etc. It is large because it also contains gcc and g++ for x86_64 for building BaseTools.
Once the Python based build tools are the default the gcc/g++
could (maybe?) move out of the base image.

The architecture-specific images then include gcc and qemu.

Reply:
Regarding the split of build and emulate, I meant to imply that we have a layered build where we add additional features on top of the base image. That is tianocore/containers/fedora-36 contains all of the tools to compile and tianocore/containers/fedora-36-qemu is based on tianocore/containers/fedora-36 but additionally have the qemu installs. That way the qemu layer can be used in cases where qemu is actually needed such as the ovmf package, but the build tools only version can be used in other cases.

Reply:
I think this is a good idea.